Q-8. The anxiety levels of 10 participants were measured before and after a new therapy. The scores are not normally distributed. Use the Wilcoxon signed-rank test to test whether the therapy had a significant effect on anxiety levels. The data is given below: Participant Before therapy After therapy Difference

In [ ]:
from IPython.display import Image
Image("Q8.png")

In [3]:
from scipy.stats import wilcoxon

In [4]:
Before_Therapy = [10, 8, 12, 15, 6, 9, 11, 7, 14, 10]
After_Therapy = [7, 6, 10, 12, 5, 8, 9, 6, 12, 8]
Difference = [-3, -2, -2, -3, -1, -1, -2, -1, -2, -2]

In [5]:
# Perform Wilcoxon signed-rank test
statistic, p_value = wilcoxon(Difference)

# Output results
print("Wilcoxon signed-rank test:")
print("Statistic:", statistic)
print("P-value:", p_value)

Wilcoxon signed-rank test:
Statistic: 0.0
P-value: 0.001953125


Observation:

The p-value obtained from the Wilcoxon signed-rank test is 0.001953125.

Since the p-value is less than the commonly used significance level of 0.05, we can conclude that the therapy had a significant effect on anxiety levels.
